utils/swift_build_support/swift_build_support/host_specific_configuration.py

```diff
@@ -203,6 +205,16 @@ def __init__(self, host_target, args):
                         "check-swift{}-optimize_none_with_implicit_dynamic-{}"
                         .format(subset_suffix, name))
 
+            # Only pull in these flags when cross-compiling with
+            # --cross-compile-hosts.
+            if deployment_target_name != args.host_target and \
+                host_target != args.host_target:
+                self.add_flags_for_cross_compilation(args, deployment_target)
+
+    def add_flags_for_cross_compilation(self, args, deployment_target):
+        self.swift_flags = deployment_target.swift_flags(args)
+        self.cmake_options = deployment_target.cmake_options(args)
```

Comment:
I spun this off as you asked for: IMO it was better as just part of the initializer above, but maybe you had some other change in mind instead.

Reply:
Its fine. I was thinking of reset_flags_for_cross_compilation or reset_deployment_target or something like that.